docs/dev/provisioning.rst

```diff
@@ -6,9 +6,289 @@ Overview
+    SECRET_KEY = get_random_string(50, chars)
+
+Then set it as DJANGO_SECRET_KEY::
+
+    $ export DJANGO_SECRET_KEY=<the random string from above>
+
+This should be different for each site being deployed, but the same on
+all servers running Django for a particular site, and not changing over
+time.
+
+Set DOMAIN to the hostname the site will be served at::
+
+    $ export DOMAIN=www.example.com
+
+The site might need to send email to admins on errors, and to users for
+things like password resets. Arrange to make an SMTP server available for
+outgoing email, then set the following Django settings as environment
+variables:
+
+* EMAIL_HOST
+* EMAIL_HOST_USER
+* EMAIL_HOST_PASSWORD
+* EMAIL_USE_TLS
+* EMAIL_USE_SSL (just set one of EMAIL_USE_TLS or EMAIL_USE_TLS to a
+  non-empty string to indicate "True"; leave the other unset)
+* EMAIL_PORT (optional; defaults to 25, 465, or 587 depending on
+  whether EMAIL_USE_TLS, EMAIL_USE_SSL, or neither are set)
+* DEFAULT_FROM_EMAIL
+* EMAIL_SUBJECT_PREFIX
+
+These are documented starting
+`here <https://docs.djangoproject.com/en/3.0/ref/settings/#email-host>`_;
+I won't bother copying the docs.
+
+Migrate and collect static
+..........................
+
+There are a couple of tasks that need to be done any time the code is
+updated, before (re)starting the server. The migrate step only needs to
+be done on one server since it updates the database that all servers are
+sharing. The collectstatic step needs to be done on every server.
+
+We generally build this into our deploy process.
+
+1. Activate the virtual env::
+
+    $ . /somepath/cc-licenses-venv/bin/activate
+
+2. Set all the environment variables (above).
+
+3. Run database migrations:
+
+    $ python manage.py migrate --noinput
+
+4. Collect all static files to STATIC_ROOT:
+
+    $ python manage.py collectstatic --noinput
+
+Run Django
+..........
+
+To get a process running Django and serving requests, we'll use a tool
+called `gunicorn <https://gunicorn.org/>`_ that's installed into the
+virtual environment.
+
+We'll run this strictly internally, listening for requests on a Unix port.
+Our web server will proxy to that port.
+
+Reminder: arrange for the environment variables mentioned above to be set
+before gunicorn is started.  (You can set them on the gunicorn command
+line with ``-e``, but it gets unwieldy.)
+
+::
+
+    $ cd path-where-we-checked-out-the-code
+    $ /somepath/cc-licenses-venv/bin/gunicorn --bind unix:/tmp/portfile cc_licenses.wsgi
+
+Gunicorn has lots of options for tuning which you can look up.
+
+Run a webserver in front
+........................
+
+We usually run nginx as our front-end web server. A simple approach is to
+add a new config file to /etc/nginx/sites-enabled for each site, making
+sure server_name is set correctly in each.  E.g.
+``/etc/nginx/sites-enabled/www.example.com.conf`` (the name is completely
+arbitrary). Then reload or restart nginx.
```

Comment:
since we show the command to start gunicorn, it might be nice to show the command to reload nginx.